integration/chaincode/simple/cmd/main.go

```diff
@@ -8,16 +8,69 @@ package main
 
 import (
 	"fmt"
+	"io/ioutil"
 	"os"
+	"os/signal"
+	"syscall"
 
 	"github.com/hyperledger/fabric-chaincode-go/shim"
 	"github.com/hyperledger/fabric/integration/chaincode/simple"
 )
 
-func main() {
-	err := shim.Start(&simple.SimpleChaincode{})
+func handleSignals(handlers map[os.Signal]func()) {
+	var signals []os.Signal
+	for sig := range handlers {
+		signals = append(signals, sig)
+	}
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, signals...)
+
+	for sig := range signalChan {
+		fmt.Fprintf(os.Stderr, "Received signal: %d (%s)", sig, sig)
+		handlers[sig]()
+	}
+}
+
+func handleSigTerm() {
+	termFile := os.Getenv("ORG1_TERM_FILE")
+	if termFile == "" {
+		termFile = os.Getenv("ORG2_TERM_FILE")
+	}
+	if termFile == "" {
+		return
+	}
+
+	err := ioutil.WriteFile(termFile, []byte("term-file"), 0644)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Exiting Simple chaincode: %s", err)
-		os.Exit(2)
+		fmt.Fprintf(os.Stderr, "failed to write term file to %s: %s", termFile, err)
+	}
+}
+
+func main() {
```

Comment:
This is the top-level function in this file. IMO, this should be at the beginning of this file. The reason is that, first, I read `handleSignals`, then I read `handleSigTerm` followed by `main()`. Finally, I had to again read `handleSignals` and `handleSigTerm` by keeping `main()` in the context. Probably, I need to improve my code reading skills. May be the rearrangement might help me.